docs/source/debugging.md

```diff
@@ -0,0 +1,183 @@
+languages. Make sure to install and enable the [Python Extension for VS Code](https://marketplace.visualstudio.com/items?itemName=ms-python.python)
+to debug hi-ml toolbox modules built in Python.
+
+### Debugging configs
+
+The hi-ml repository is organised in
+[Multi-root workspaces](https://code.visualstudio.com/docs/editor/workspaces#_multiroot-workspaces) to account
+for environment differences among the modules and offer flexibility to configure each module seperatly.
+
+We provide a set of example debugging configs for each of hi-ml module:
+
+* [launch.json in hi-ml](https://github.com/microsoft/hi-ml/tree/main/hi-ml/.vscode/lanch.json)
+* [launch.json in hi-ml-azure](https://github.com/microsoft/hi-ml/tree/main/hi-ml-azure/.vscode/lanch.json)
+* [launch.json in hi-ml-histopathology](https://github.com/microsoft/hi-ml/tree/main/hi-ml-histopathology/.vscode/lanch.json)
+
+VS Code restricts debugging to user-written code only by default. If you want to step through external code and
+standard libraries functions, set `"justMyCode": false` inside the debugging config block in the `launch.json` file.
+
+In particular, if you would like to debug the current file while breaking through `pytorch` code, navigate to
+`himl-projects.code-workspace` in the repo root and edit the "launch" block as follow:
+
+```json
+"launch": {
+    "configurations": [
+      {
+        "name": "Python: Current File",
+        "type": "python",
+        "request": "launch",
+        "program": "${file}",
+        "console": "integratedTerminal",
+        "justMyCode": false
+      },
+    ],
+}
+```
+
+### Pytorch Lightning flags for debugging and quick runs
+
+The hi-ml toolbox is built upon [Pytorch Lightning (PL)](https://www.pytorchlightning.ai/) to help you build scalable
+deep learning models for healtcare and life sciences. Refer to [Running ML Experiments with hi-ml](runner.md) for
+detailed instructions on how to build scalable pipelines within hi-ml.
+
+Whether you're building a brand new model, or extending an existing one, you might want to make sure that your
+code runs as expected locally before submitting a job to AzureML. hi-ml supports a set of debugging flags that triggers
+[Pytorch Lightning Trainer](https://pytorch-lightning.readthedocs.io/en/stable/common/trainer.html#trainer) arguments to
+help you detect any potential errors or bugs at early stage.
+
+These are available as part of the
+[TrainerParams](https://github.com/microsoft/hi-ml/blob/96b8ba60ebf84416f5c9b13e2df15ee07a13e6bb/hi-ml/src/health_ml/deep_learning_config.py#L357)
+and can be used as extra command line arguments with the [hi-ml
+runner](https://github.com/microsoft/hi-ml/blob/746c8b58c1af71f71eeaaac2a8584be1d9a5386f/hi-ml/src/health_ml/runner.py#L107).
+
+* `pl-fast-dev-run`: If set to `n`, runs the pipeline for only `n` batch(es) of train, val and test for only a single
```

Comment:
Can you double check, I think the separator for the arg is underscore `_` rather than dash `-`?

Reply:
You're right!